pkg/visor/init.go

```diff
@@ -186,10 +187,39 @@ func initSNet(ctx context.Context, v *Visor, log *logging.Logger) error {
 		STCP: v.conf.STCP,
 	}
 
+	var nat stun.NATType
+	var host *stun.Host
+	var err error
+stunLoop:
```

Comment:
It is somewhat annoying that this is in `snet` package as the Stun check takes a substantial amount of time and blocks most other modules from initializing with it. @i-hate-nicknames does your refactor make it possible to move this to some other module that is not the main dependency for most modules?

Reply:
@jdknives Yes I think. Looks to me as the following:
 
1. Make a stun module similarly to other modules
2. Make some stun "fetcher" object that will that can initialize itself with this heavy logic and then store results.
3. Create init function for the module in (1). Create object from (2) there and call the heavy logic. Then assign this object into visor field.

Similarly to what other modules do essentially.

Reply:
@ersonp can we avoid using goto though?

Reply:
I'll try to find a better way without using goto. Let me know if you have any suggestions @i-hate-nicknames